src/NzbDrone.Core/Download/Clients/rTorrent/RTorrentProxy.cs

```diff
@@ -0,0 +1,212 @@
+
+        [XmlRpcMethod("load_raw_start")]
+        int LoadBinary(byte[] data);
+
+        [XmlRpcMethod("d.erase")]
+        int Remove(string hash);
+
+        [XmlRpcMethod("d.set_custom1")]
+        string SetLabel(string hash, string label);
+
+        [XmlRpcMethod("d.set_priority")]
+        int SetPriority(string hash, int priority);
+
+        [XmlRpcMethod("d.get_name")]
+        string GetName(string hash);
+
+        [XmlRpcMethod("system.client_version")]
+        string GetVersion();
+    }
+
+    public class RTorrentProxy : IRTorrentProxy
+    {
+        private readonly Logger _logger;
+
+        public RTorrentProxy(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public string GetVersion(RTorrentSettings settings)
+        {
+            var client = BuildClient(settings);
+            var version = client.GetVersion();
+            return version;
+        }
+
+        public List<RTorrentTorrent> GetTorrents(RTorrentSettings settings)
+        {
+            var client = BuildClient(settings);
+            var ret = client.TorrentMulticall("main",
+                "d.get_name=", // string
+                "d.get_hash=", // string
+                "d.get_base_path=", // string
+                "d.get_size_bytes=", // int
+                "d.get_left_bytes=", // int
+                "d.get_down_rate=", // int (in bytes / s)
+                "d.get_custom1=", // string (label) 
+                "d.is_open=", // int
+                "d.is_active=", // int
+                "d.get_complete=", //int
+                "d.get_priority="); //int
+
+            var items = new List<RTorrentTorrent>();
+            foreach (System.Object[] torrent in ret)
+            {
+                var category = (String)torrent[6];
+                // Don't concern ourselves with categories other than specified
+                if (category != settings.TvCategory) continue;
+
+                var item = new RTorrentTorrent();
+                item.Name = (String)torrent[0];
+                item.Hash = (String)torrent[1];
+                item.BaseDir = (String)torrent[2];
+                item.TotalSize = (Int64)torrent[3];
+                item.RemainingSize = (Int64)torrent[4];
+                item.Category = category;
+                item.IsOpen = Convert.ToBoolean((Int64)torrent[7]);
+                item.IsActive = Convert.ToBoolean((Int64)torrent[8]);
+                item.IsFinished = Convert.ToBoolean((Int64)torrent[9]);
+
+                var down_rate = (Int64)torrent[5];
+                item.RemainingTime = (down_rate > 0 && item.IsActive) ? (item.RemainingSize / down_rate) : -1;
+
+                items.Add(item);
+            }
+            return items;
+        }
+
+        /*
+         * It is only ready if:
+         *  - It exists
+         *  - The name does not match "{$hash}.meta"
+         */
+        public bool IsHashReadyTorrent(String hash, RTorrentSettings settings)
```

Comment:
Remove comment and change method to something _like_ HasMagnetLinkResolved, not sure yet, but that way you don't need a comment.